data_structures/linked_list/linked_list.py

```diff
@@ -0,0 +1,62 @@
+from node import Node
+
+class LinkedList:
+    """class for new linked lists"""
+    def __init__(self, iter = []):
+        self.head = None
+        self._len = 0 
+
+        for item in iter:
+            self.head = Node(item, self.head)
+            self._len += 1
+
+    
+
+    def __len__(self):
+        """return len of the corrent object"""
+        return self._len
+    
+    
+    def __str__(self):
+        """return all items from the LL"""
+        lis = []
```

Comment:
This method will not work properly the str magic method can only return strings (run your code and you would find this).